ChopChop/Models/Ingredients/IngredientCategory.swift

```diff
@@ -0,0 +1,87 @@
+import GRDB
+import Combine
+
+/**
+ Represents a collection of ingredients grouped under a category.
+ 
+ Invariants:
+ - The `ingredientCategoryId` of all ingredients contained in this category is the same as the category's `id`.
+ */
+class IngredientCategory: FetchableRecord {
+    var id: Int64?
+    private(set) var name: String
+    private(set) var ingredients: [Ingredient]
+
+    init(name: String, id: Int64? = nil, ingredients: [Ingredient] = []) {
+        self.id = id
+        self.name = name
+
+        ingredients.forEach { $0.ingredientCategoryId = id }
+        self.ingredients = ingredients
+    }
+
+    required init(row: Row) {
+        id = row["id"]
+        name = row["name"]
+        ingredients = []
+
+        if let categoryId = id {
+            let storageManager = StorageManager()
+            storageManager.ingredientsFilteredByCategoryOrderedByNamePublisher(ids: [categoryId])
+                .sink(receiveCompletion: { _ in }, receiveValue: { [weak self] infos in
+                    let ingredientIds = infos.compactMap { $0.id }
+                    let ingredients = ingredientIds.compactMap { try? storageManager.fetchIngredient(id: $0) }
+                    self?.ingredients = ingredients
+                })
+        }
```

Comment:
This is actually not an intended use of the publisher. The publishers are intended to be used in the view models to update the list of objects to be rendered. If you want I can either

1. Provide a method to fetch ingredients given a categoryId
2. Load the ingredients together with the category to begin with so you can avoid doing this

Reply:
I think 1. would be good